projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/subscribe-overlay.js

```diff
@@ -0,0 +1,59 @@
+const { domReady } = wp;
+
+domReady( function () {
+	const overlay = document.querySelector( '.jetpack-subscribe-overlay' );
+	const overlayDismissedCookie = 'jetpack_post_subscribe_overlay_dismissed';
+	const hasOverlayDismissedCookie =
+		document.cookie && document.cookie.indexOf( overlayDismissedCookie ) > -1;
+
+	if ( ! overlay || hasOverlayDismissedCookie ) {
+		return;
+	}
+
+	const close = document.querySelector( '.jetpack-subscribe-overlay__close' );
+	close.onclick = function ( event ) {
+		event.preventDefault();
+		closeOverlay();
+	};
+
+	const toContent = document.querySelector( '.jetpack-subscribe-overlay__to-content' );
+	// User can edit overlay, and could remove to content link.
+	if ( toContent ) {
+		toContent.onclick = function ( event ) {
+			event.preventDefault();
+			closeOverlay();
+		};
+	}
+
+	const form = document.querySelector( '.jetpack-subscribe-overlay form' );
+	if ( form ) {
+		form.addEventListener( 'submit', closeOverlay );
+	}
+
+	function closeOverlayOnEscapeKeydown( event ) {
+		if ( event.key === 'Escape' ) {
+			closeOverlay();
+		}
+	}
+
+	function openOverlay() {
+		overlay.classList.add( 'open' );
+		document.body.classList.add( 'jetpack-subscribe-overlay-open' );
```

Comment:
a bit nit picky but would it make sence to use 
`jetpack-subscribe-overlay-is-open` class here?  and the same with above? `is-open`

Reply:
This is copy&paste from the Subscribe Modal:
https://github.com/Automattic/jetpack/blob/9393263c72c87a4143c6efb68de6f8547ca4df97/projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.js#L49-L50

I mean, I could change it but don't see the point. Leaving it like this we are at least keeping the naming convention with the modal.

Reply:
Yeah that make sense.